### DISCUSSION TITLE
Meeting: August 13th, 2024 agenda

### DISCUSSION THREAD

**Discussion by earth2marsh** [OAI/sig-moonwalk#148] [Announcements]
## Weekly meetings happen on Tuesdays

**Time:** 5pm GMT / 9am Pacific
**Zoom:** [https://zoom.us/j/975841675](https://zoom.us/j/975841675?pwd=SUh4MjRLaEFKNlI3RElpWTdhRDVVUT09), dial-in passcode: 763054

### Accessibility & Etiquette

* Participants must abide by our [Code-of-Conduct](https://github.com/OAI/OpenAPI-Specification/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
* Meetings are recorded for future reference, and for those who are not able to attend in-person.
* We invited challenges to language or behaviour that is harmful or not inclusive during this meeting.
* We look forward to your participation, but please consider these acts of etiquette:
  * Remain on mute when not speaking to prevent interruptions.
  * Blur your background to reduce visual distractions.
  * Use the Zoom meeting "Raise Hand" feature to notify the group when you wish to speak.

| Blur My Background | Raise Hand |
|-|-|
| <img width="323" alt="Screenshot of Zoom UI showing the 'Stop Video' and 'Blur My Background' control" src="https://github.com/OAI/OpenAPI-Specification/assets/7367/7e43dbbb-6529-46e6-8b04-4c1aa852d9dd"> | <img width="323" alt="Screenshot of Zoom UI showing the 'Reaction' and 'Raise Hand' control" src="https://github.com/OAI/OpenAPI-Specification/assets/7367/f991722f-4651-40aa-9bc4-7e9a2a165a6a"> |

### Agenda Structure

| Topic | Owner | Decision/NextStep |
|-------|---------|---------|  
| | | |
Intros and governance meta-topics (5 mins) | SIG members | -- |
General discussion topics (list below) | SIG members | -- |
Wrap-up and seed the next agenda (5 mins) | SIG members | -- |

Whether attending or not, **anyone can comment on this issue prior to the meeting to suggest topics or to add comments on planned topics or proposals**.

**Comment by handrews** [comment]
Let's continue the data modeling discussion with high level thoughts on SDK vs HTTP interfaces.  Background:
* [July 30th agenda](https://github.com/OAI/sig-moonwalk/discussions/146#discussioncomment-10184006) (we discussed all of this)
* [Aug 6th agenda](https://github.com/OAI/sig-moonwalk/discussions/147#discussioncomment-10246873) (we only discussed the very first bullet point group so far)

Here are the "for next week" comments from last week's [agenda](https://github.com/OAI/sig-moonwalk/discussions/147):

* Philosophical discussion about the relative importance of the HTTP-side interface vs the SDK-side interface
    * Is one more important?  To whom?
    * How does this fit with how tools are currently written and used?
    * Background philosophical concept: consumer over producer (but consumer of HTTP or consumer of SDK?)
* Scope:  XML and other less-common targets
    * many older / more slow-moving standards are still XML-centric
    * [Extensibility](https://github.com/OAI/OpenAPI-Specification/discussions/3771)?  It's easier if we don't have to solve all target representation problems up front
    * Note the [example of one schema mapping to many targets in the earlier comment](https://github.com/OAI/sig-moonwalk/discussions/147#discussioncomment-10246885)- do we need this?
    * One SDK representatoin for multiple representation at the HTTP may make model-to-target mappings more clear
* Consider the use case groups ([client, sever, tools- Gary to elaborate on distinctions](https://github.com/OAI/sig-moonwalk/discussions/147#discussioncomment-10256131)) in the above comment for any specific data modeling features
    * We'll probably get back to this after the higher-level SDK/HTTP discussions resolve a bit

**Reply by earth2marsh** [reply]
We'll carry this over to next week!

**Comment by earth2marsh** [comment]
Today we began with some discussion of URI-encoding and why OpenAPI inherits opinions from its HTTP-orientation. 

We then discussed how having an ontology and taxonomy might help organize the rest of the spec work (as well as helping us arrive at some common language!). I think @handrews has done some work in the last 6 months in this space? I also had started some draft definitions, and I walked through some of those. There was general support for more Moonwalk definitions, so please add suggestions and new words to define (and even definitions!) [here](https://docs.google.com/document/d/15eZQxSflO5SNopUvhx6pjQH0EB2n5BslpnRTSs0F1LY/edit). 

[Video](https://zoom.us/rec/share/bY2pqUtE24-QTLSms9CV9vxMCourGw4Z_ZzX8C1ziEII4CnAiiYiVCj7J-4H2Oxj.VfbrxkXWDbMpiHJP) has been posted.
